pkg/distributor/validate.go

```diff
@@ -387,9 +399,17 @@ func validateLabels(m *sampleValidationMetrics, cfg labelValidationConfig, userI
 	}
 
 	if !skipLabelCountValidation && len(ls) > cfg.MaxLabelNamesPerSeries(userID) {
```

Comment:
I think you should move the `len(ls) > cfg.MaxLabelNamesPerSeries(userID)` check here, since otherwise it semi overrides `cfg.MaxLabelNamesPerInfoSeries(userID)`. It's better if the two limits are independent of each other (even though generally `MaxLabelNamesPerInfoSeries` should be higher).

Reply:
I think it makes sense this way. If we renamed MaxLabelNamesPerSeries to MaxLabelNamesPerSeriesExceptInfoSeries then it would be independent, but I think the naming constrains us a little. But thanks for point this out, I'll update the help text!